tests/test_openeo_gfmap/test_unit_patch_extractors.py

```diff
@@ -0,0 +1,91 @@
+from openeo_gfmap.features import PatchFeatureExtractor
+from openeo.udf import XarrayDataCube
+
+import pytest
+import numpy as np
+import xarray as xr
+from unittest.mock import MagicMock, patch
+
+LAT_HARMONIZED_NAME = "GEO-LAT"
+LON_HARMONIZED_NAME = "GEO-LON"
+EPSG_HARMONIZED_NAME = "GEO-EPSG"
+
+class DummyPatchFeatureExtractor(PatchFeatureExtractor):
+    def output_labels(self):
+        return ["label1", "label2"]
+
+    def execute(self, inarr: xr.DataArray) -> xr.DataArray:
+        return inarr  # Simplified for testing purposes
+
+@pytest.fixture
+def mock_feature_extractor():
+    return DummyPatchFeatureExtractor()
+
+@pytest.fixture
+def mock_data_array():
+    return xr.DataArray(np.random.rand(10, 10), dims=["y", "x"])
```

Comment:
Using random values in tests is bad practice as you might encounter edge cases that you later on can't reproduce. I would rather use @pytest.mark.parametrize to create multiple test cases. (See the python-client for some examples)